### DISCUSSION TITLE
Pull Request feature needs multiple improvements

### DISCUSSION THREAD

**Discussion by chriskuech** [community/community#9816] [Pull Requests]
Pull Requests are taking significantly more of my time than with Azure DevOps due to a few very tedious interaction points.  I hope these can be addressed.

- [ ] https://github.com/github/feedback/discussions/8531
- [ ] Possibly https://github.com/github/feedback/discussions/8597
- [ ] Pull Request flow is very unintuitive.  I understand it now and it works ok but there was definitely a learning curve.  Namely it is very confusing that PRs are a point in time, rather than Azure DevOps where a review is just setting the approval state of the PR and adding comments.  Most confusingly, comments can be added both to a discrete review as well as to the PR itself.
- [ ] Pull Request comments can be shown as "outdated".  This is very obnoxious to me and has cost me much time clicking through GitHub.  Most often, these comments are not "outdated" as much as the change or file addition has been completely removed from the PR, yet you must manually look at the latest version of the PR and check for the change to figure this out.  The PR comment should should include a toggle allowing you to see the diff between target branch and commented commit, but more importantly it should allow showing the diff between target branch and latest commit, clearly indicating whether the file has been deleted or if there are no longer changes between target branch and source branch.
- [ ] The Diff Span Indicator (below) never conveys helpful information.  I am never interested in the number of commits I'm looking at.  I'm much more interested whether the starting commit is the branching point and whether the ending commit is the latest commit or outdated commit.  If I'm not looking at the latest commit, I want a large visual indicator (ex: yellow banner) especially when I'm scrolled away from the Diff Span Indicator.
    ![image](https://user-images.githubusercontent.com/3680671/148585338-0b9ac29a-0355-4775-ab2d-17ec8e695f45.png)

**Comment by hazz0003** [comment]
> Pull Requests are taking significantly more of my time than with Azure DevOps due to a few very tedious interaction points.

My company recently made the switch from Azure DevOps to GitHub.  Our time on getting PR's completed has increased dramatically.  I can't quantify in minutes; but I can *Objectively* say we have a lower sprint velocity over several sprints now and we spend more time on PRs. We have a column in our KanBan board dedicated to PR's.  It's gone from maybe a few stories stuck in that column to tens of items consistently.  

I'd say our review quality has gone down too.  After first, though this was learning pains of the new tooling.  I've changed my mind, there are obvious gaps some of which @chriskuech identified above.  Life at Github.com is harder than it needs to be:  

> Namely it is very confusing that PRs are a point in time, rather than Azure DevOps where a review is just setting the approval state of the PR and adding comments. Most confusingly, comments can be added both to a discrete review as well as to the PR itself.

@chriskuech  I couldn't agree more, I am struggling to find a workflow where I would want a comment in a discrete review.  If anything that should be an exceptional case.
